query/src/main/scala/filodb/query/exec/RangeVectorTransformer.scala

```diff
@@ -368,10 +369,16 @@ final case class AbsentFunctionMapper(columnFilter: Seq[ColumnFilter], rangePara
 
     val resultRv = nonNanTimestamps.map { t =>
       val rowList = new ListBuffer[TransientRow]()
-      for (i <- rangeParams.startSecs to rangeParams.endSecs by rangeParams.stepSecs) {
-        if (!t.contains(i * 1000))
-          rowList += new TransientRow(i * 1000, 1)
-      }
+
+      val it = Iterator.from(0, rangeParams.stepSecs.toInt)
+        .takeWhile(_ <= rangeParams.endSecs - rangeParams.startSecs).map { i =>
+        val timestamp = i + rangeParams.startSecs
+        if (!t.contains(timestamp * 1000))
+          rowList += new TransientRow(timestamp * 1000, 1)
+       }
+
+      // address step == 0 case
+      if (rangeParams.startSecs == rangeParams.endSecs) it.take(1).toList else it.toList
```

Comment:
potential infinite loop here, can you add `step == 0` condition also

Reply:
Done